### DISCUSSION TITLE
a little Gantt Chart experiment

### DISCUSSION THREAD

**Discussion by EdoFro** [freeplane/freeplane#1121] [Show and tell]
Hi all,
I tried to build a Gantt chart with the new node layouts of Freeplane v1.11.1.

I wrote some very basic scripts, so it sets the layout automatically. (sets and unsets)

Here some images of what I have reached so far:

mindmap with tasks data as attributes:
![image](https://user-images.githubusercontent.com/35700575/235745108-a0728828-0005-40bd-96b5-b1e7a1189f10.png)

![image](https://user-images.githubusercontent.com/35700575/235745188-bfd5f983-85b9-46af-8560-c26e8372d39d.png)

Applying Gantt Layout:
![image](https://user-images.githubusercontent.com/35700575/235745479-76bf8870-4741-47e7-9c0b-38c61263a9d4.png)

![](https://media.tenor.com/xU1mDhSARNcAAAAd/dance-lonely-boy.gif)

**Comment by Illegit** [comment]
Hi EdoFro, I find interesting your gantt exercise, may I suggest having a look to my map here in the attachment: this map lacks the gantt but utilizes some of the tools suggested to me by you, by macmarrum and others. My most important tool is to have all of the characteristics of the mindmap tool + clones + attributes + branches&leaves, combined in a WBS (fundamental) and a customizable TIMELINE (here in the example I use it only for MILESTONES). It could be useful to have even the possibility to transform the WBS in gantt and back (as you show). What do you think?
[WBS timeline.zip](https://github.com/freeplane/freeplane/files/11390960/WBS.timeline.zip)
![Screenshot 2023-05-03 220008](https://user-images.githubusercontent.com/62674157/236041275-4162076c-492d-4d42-8f8e-a128d0a983f0.png)

**Reply by EdoFro** [reply]
I haven't had time to look at this topic again. I was away for a week and now I'm trying to fullfill my tasks at work.
This is still in my pending tasks bucket ad I'm interested in work on it further.

**Comment by Illegit** [comment]
Hi, EdoFro may you share how did you create the gantt?
I like to experiment with it.
Thanks

**Reply by EdoFro** [reply]
I sent you an invitation to access the repo that I will use to develop a Gantt Chart add-on.

look at the mindmap [Gantt layout example.mm](https://github.com/EdoFro/Freeplane_Gantt-o-Matic/blob/main/Gantt%20layout%20example.mm). it has the scripts that I launch using the **Menu-o-Matic** add-on

**Reply by EdoFro** [reply]
@Illegit,
I created a little "How to" [here](https://github.com/EdoFro/Freeplane_Gantt-o-Matic/discussions/1)
any feedback is welcomed

**Reply by greg4b** [reply]
the link goes to "404" page 

Please, provide "How to" - thanks

**Comment by johnWcarey** [comment]
EdoFro, I am very interested in what you have done with respect to a Gantt Chart.  I'd love to take a look at your repo to get an idea of how you are doing and see if I might be able to contribute.

**Reply by EdoFro** [reply]
@johnWcarey,
There is little information, but I can answer any question you have.

I'm still interested in develop this as an addon, but I decided to focus in ending other addons first.
- [Menu-o-Matic](https://github.com/EdoFro/Freeplane_Menu-o-Matic)
- [TabbedPanelMod](https://github.com/EdoFro/Freeplane_TabbedPanelMod)

The first one lacks the documentation to be ready.
The second one, I consider to be ready.

That is why I hope to be able to resume this project in the near future.

**Comment by EdoFro** [comment]
@greg4b, @johnWcarey,
Now the [repository ](https://github.com/EdoFro/Freeplane_Gantt-o-Matic) is public.

It was private because the project is in a very early state (not even alpha). It's more like a series of tests or proofs of concepts rather than a development.

There is some more information about it in its "Discussion" forum.

At the moment it is based on the use of script formulas, but the idea is to make an addon that recalculates durations, start and end dates using nodeChangeListeners API.

The idea is also that it has a panel in the ToolPanel with buttons and fields to define tasks and dependants.

@Illegit has been helping me in the discussion and definition of this feature.

**Reply by EdoFro** [reply]
Oh, oh. all images from https://github.com/EdoFro/Freeplane_Gantt-o-Matic/discussions/1 got lost when I made the repository Public.

I will make it private again and send you invitations.

**Comment by Illegit** [comment]
I confirm it is pretty interesting. And I strongly suggest to utilize the EdoFro addons listed above: 
[Menu-o-Matic](https://github.com/EdoFro/Freeplane_Menu-o-Matic)
[TabbedPanelMod](https://github.com/EdoFro/Freeplane_TabbedPanelMod)
